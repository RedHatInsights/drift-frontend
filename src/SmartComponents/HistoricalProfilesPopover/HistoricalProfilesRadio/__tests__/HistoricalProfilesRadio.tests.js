import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';

import ConnectedHistoricalProfilesRadio, { HistoricalProfilesRadio } from '../HistoricalProfilesRadio';

/*eslint-disable camelcase*/
describe('HistoricalProfilesRadio', () => {
    let props;

    beforeEach(() => {
        props = {
            selectedHSPIds: [],
            profile: { id: '1234', captured_date: '03 May 2020, 18:20 UTC' }
        };
    });

    it('should render correctly', () => {
        const wrapper = shallow(
            <HistoricalProfilesRadio { ...props }/>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render mount correctly', () => {
        const wrapper = mount(
            <HistoricalProfilesRadio { ...props }/>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe('ConnectedHistoricalProfilesRadio', () => {
    let initialState;
    let props;
    let mockStore;

    beforeEach(() => {
        mockStore = configureStore();
        initialState = {
            entities: {}
        };

        props = {
            profile: { id: '1234', captured_date: '03 May 2020, 18:20 UTC' },
            selectedHSPIds: []
        };
    });

    it('should render correctly', () => {
        const store = mockStore(initialState);
        const wrapper = mount(
            <MemoryRouter keyLength={ 0 }>
                <Provider store={ store }>
                    <ConnectedHistoricalProfilesRadio { ...props } />
                </Provider>
            </MemoryRouter>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
/*eslint-enable camelcase*/
