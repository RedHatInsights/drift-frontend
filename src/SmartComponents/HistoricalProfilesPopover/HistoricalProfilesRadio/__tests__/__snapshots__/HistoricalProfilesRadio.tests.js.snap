// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedHistoricalProfilesRadio should render correctly 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Connect(HistoricalProfilesRadio)
        profile={
          Object {
            "captured_date": "03 May 2020, 18:20 UTC",
            "id": "1234",
          }
        }
        selectedHSPIds={Array []}
      >
        <HistoricalProfilesRadio
          dispatch={[Function]}
          entities={Object {}}
          profile={
            Object {
              "captured_date": "03 May 2020, 18:20 UTC",
              "id": "1234",
            }
          }
          selectedHSPIds={Array []}
        >
          <Radio
            className=""
            id="03 May 2020, 18:20 UTC"
            isChecked={false}
            isDisabled={false}
            isValid={true}
            label="03 May 2020, 18:20 UTC"
            name="03 May 2020, 18:20 UTC"
            onChange={[Function]}
            value="03 May 2020, 18:20 UTC"
          >
            <div
              className="pf-c-radio"
            >
              <input
                aria-invalid={false}
                checked={false}
                className="pf-c-radio__input"
                data-ouia-component-id="OUIA-Generated-Radio-2"
                data-ouia-component-type="PF4/Radio"
                data-ouia-safe={true}
                disabled={false}
                id="03 May 2020, 18:20 UTC"
                name="03 May 2020, 18:20 UTC"
                onChange={[Function]}
                type="radio"
                value="03 May 2020, 18:20 UTC"
              />
              <label
                className="pf-c-radio__label"
                htmlFor="03 May 2020, 18:20 UTC"
              >
                03 May 2020, 18:20 UTC
              </label>
            </div>
          </Radio>
        </HistoricalProfilesRadio>
      </Connect(HistoricalProfilesRadio)>
    </Provider>
  </Router>
</MemoryRouter>
`;

exports[`HistoricalProfilesRadio should render correctly 1`] = `
<Fragment>
  <Radio
    className=""
    id="03 May 2020, 18:20 UTC"
    isChecked={false}
    isDisabled={false}
    isValid={true}
    label="03 May 2020, 18:20 UTC"
    name="03 May 2020, 18:20 UTC"
    onChange={[Function]}
    value="03 May 2020, 18:20 UTC"
  />
</Fragment>
`;

exports[`HistoricalProfilesRadio should render mount correctly 1`] = `
<HistoricalProfilesRadio
  profile={
    Object {
      "captured_date": "03 May 2020, 18:20 UTC",
      "id": "1234",
    }
  }
  selectedHSPIds={Array []}
>
  <Radio
    className=""
    id="03 May 2020, 18:20 UTC"
    isChecked={false}
    isDisabled={false}
    isValid={true}
    label="03 May 2020, 18:20 UTC"
    name="03 May 2020, 18:20 UTC"
    onChange={[Function]}
    value="03 May 2020, 18:20 UTC"
  >
    <div
      className="pf-c-radio"
    >
      <input
        aria-invalid={false}
        checked={false}
        className="pf-c-radio__input"
        data-ouia-component-id="OUIA-Generated-Radio-1"
        data-ouia-component-type="PF4/Radio"
        data-ouia-safe={true}
        disabled={false}
        id="03 May 2020, 18:20 UTC"
        name="03 May 2020, 18:20 UTC"
        onChange={[Function]}
        type="radio"
        value="03 May 2020, 18:20 UTC"
      />
      <label
        className="pf-c-radio__label"
        htmlFor="03 May 2020, 18:20 UTC"
      >
        03 May 2020, 18:20 UTC
      </label>
    </div>
  </Radio>
</HistoricalProfilesRadio>
`;
